Utils/expect_common.dart

```diff
@@ -401,7 +401,10 @@ final bool assertStatementsEnabled = (() {
 })();
 
 /// Is true iff js compiler is used
-final bool isJS = identical(1.0, 1);
+const bool isJS = identical(1.0, 1);
```

Comment:
nit: For a general purpose check, check these. I believe the common advice is to check the specific semantic differences you care about for a test, but for a general purpose check, this is likely cleaner.

> const bool isDDC = const bool.fromEnvironment('dart.library._ddc_only');
> const bool isDart2JS = const bool.fromEnvironment('dart.tool.dart2js');
> const bool isJS = isDart2JS || isDDC;